libs/markets/src/lib/components/market-info/market-info-accordion.tsx

```diff
@@ -88,14 +88,31 @@ export const MarketInfoAccordion = ({
     market.accountsConnection?.edges
   );
 
-  const settlementData = market.tradableInstrument.instrument.product
-    .dataSourceSpecForSettlementData.data as DataSourceDefinition;
-  const terminationData = market.tradableInstrument.instrument.product
-    .dataSourceSpecForTradingTermination.data as DataSourceDefinition;
+  const settlementData =
+    'dataSourceSpecForSettlementData' in
```

Comment:
Again, I think all these conditions should be based on product type

Reply:
updated here